python/taichi/lang/_wrap_inspect.py

```diff
@@ -0,0 +1,164 @@
+import atexit
+import inspect
+import os
+import tempfile
+
+import sourceinspect
+
+_builtin_getfile = inspect.getfile
+_builtin_findsource = inspect.findsource
+
+use_sourceinspect = int(os.getenv('USE_SOURCEINSPECT', 0)) == 1
```

Comment:
IIUC now you can add a test in test_deprecations.py?

Reply:
As the variable `use_sourceinspect` is determined at the time when taichi is imported, I think it's not that straight-forward to add such a test in `test_deprecations.py`?

Reply:
you can set the env variable...

Reply:
I don't understand why the test function doesn't raise the expected warning message here ...